hca/dss/__init__.py

```diff
@@ -37,24 +38,21 @@ def download(self, bundle_uuid, replica, version="", dest_name=""):
             file_uuid = file_["uuid"]
             filename = file_.get("name", file_uuid)
 
-            logger.info("File %s: Retrieving...", filename)
-            response = self.get_file._request(dict(uuid=file_uuid, replica=replica), stream=True)
+            logger.info("Retrieving file %s", filename)
 
+            file_path = os.path.join(dest_name, filename)
+
+            retries = self.get_session().adapters["https://"].max_retries
+            # When streaming response data, requests/urllib3 does not obey its usual retry logic, so we reenact it here.
             try:
-                if response.ok:
-                    file_path = os.path.join(dest_name, filename)
+                with self.get_file.stream(uuid=file_uuid, replica=replica) as fh, open(file_path, "wb") as dest_fh:
                     logger.info("%s", "File {}: GET SUCCEEDED. Writing to disk.".format(filename))
-                    with open(file_path, "wb") as fh:
-                        for chunk in response.iter_content(chunk_size=1024*1024):
-                            if chunk:
-                                fh.write(chunk)
+                    while True:
+                        dest_fh.write(fh.raw.read(1024*1024))
```

Comment:
I assume you want to terminate this loop at some point.

Reply:
The loop terminates when `retries.increment()` raises an exception when the retry count is exhausted.

Reply:
Oops. Never mind... fixing.

Reply:
Fixed. Still new at this programming thing.